AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -472,74 +477,26 @@ private static void deleteSentRequests(int requestsCount) {
     if (requestsCount == 0) {
       return;
     }
-
-    synchronized (lock) {
-      Context context = Leanplum.getContext();
-      SharedPreferences preferences = context.getSharedPreferences(
-          LEANPLUM, Context.MODE_PRIVATE);
-      SharedPreferences.Editor editor = preferences.edit();
-      int start = preferences.getInt(Constants.Defaults.START_COUNT_KEY, 0);
-      int newStart;
-      if (MAX_KEY_COUNT - start - requestsCount >= 0) {
-        for (int i = start; i < start + requestsCount; i++) {
-          editor.remove(String.format(Locale.US, Constants.Defaults.ITEM_KEY, i));
-        }
-        newStart = start + requestsCount;
-      } else {
-        for (int i = start; i < MAX_KEY_COUNT; i++) {
-          editor.remove(String.format(Locale.US, Constants.Defaults.ITEM_KEY, i));
-        }
-        for (int i = 0; i < start - MAX_KEY_COUNT + requestsCount; i++) {
-          editor.remove(String.format(Locale.US, Constants.Defaults.ITEM_KEY, i));
-        }
-        newStart = start - MAX_KEY_COUNT + requestsCount;
-      }
-
-      editor.putInt(Constants.Defaults.START_COUNT_KEY, newStart);
-      SharedPreferencesUtil.commitChanges(editor);
+    synchronized (Request.class) {
+      LeanplumEventDataManager.deleteEvents(requestsCount);
     }
   }
 
   static List<Map<String, Object>> popUnsentRequests() {
-    return getUnsentRequests(true);
-  }
-
-  static List<Map<String, Object>> getUnsentRequests() {
-    return getUnsentRequests(false);
+    return getUnsentRequests();
   }
 
-  private static List<Map<String, Object>> getUnsentRequests(boolean remove) {
-    List<Map<String, Object>> requestData = new ArrayList<>();
+  private static List<Map<String, Object>> getUnsentRequests() {
+    List<Map<String, Object>> requestData;
 
-    synchronized (lock) {
+    synchronized (Request.class) {
```

Comment:
commit looks fine, i'm just worried do we really need `synchronized (Request.class)` ?